app/src/main/java/com/kodeco/android/countryinfo/ui/screens/countryinfo/CountryInfoScreen.kt

```diff
@@ -20,17 +22,31 @@ fun CountryInfoScreen(
     onCountryClicked: (String) -> Unit,
     navigateToAboutScreen: () -> Unit,
 ) {
-
     val message = stringResource(R.string.something_went_wrong)
     val state by countryInfoViewModel.uiState.collectAsState()
 
-    val onReload = {
-        countryInfoViewModel.processIntent(CountryInfoIntent.LoadCountries)
+    val featureEnabled by countryInfoViewModel.featureEnabled.collectAsState(false)
+
+    val offlineEnabled by countryInfoViewModel.offlineEnabled.collectAsState()
+    val favoriteEnabled by countryInfoViewModel.favoriteEnabled.collectAsState()
+
+    val intent = when {
+        favoriteEnabled -> CountryInfoIntent.LoadFavoriteCountries
+        offlineEnabled -> CountryInfoIntent.LoadCountriesFromLocalStorage
+        else -> CountryInfoIntent.LoadCountries
+    }
+
+    // updates ui each time intent changes (offline mode works much better)
+    LaunchedEffect(intent) {
+        delay(500)
+        countryInfoViewModel.processIntent(intent)
     }
```

Comment:
I would caution a bit against having all this logic here.  This likely should go into the viewmodel.  Normally, views and UI shouldn't rely on this kind of logic.    This is more business logic than UI logic.